Sources/Controllers/ALKConversationListTableViewController.swift

```diff
@@ -119,7 +119,91 @@ public class ALKConversationListTableViewController: UITableViewController, Loca
         }
         tableView.estimatedRowHeight = 0
+        guard configuration.enableDeleteConversationOnLongpress else{ return }
+        let longPress = UILongPressGestureRecognizer(target: self, action: #selector(longPress))
+        tableView.addGestureRecognizer(longPress)
+    }
+    
+    @objc func longPress(sender: UILongPressGestureRecognizer) {
+        if sender.state == UIGestureRecognizer.State.began {
+            let touchPoint = sender.location(in: tableView)
+            if let indexPath = tableView.indexPathForRow(at: touchPoint) {
+                // your code here, get the row for the indexPath or do whatever you want
+                if viewModel.chatFor(indexPath: indexPath) != nil, let conversation = viewModel.getChatList()[indexPath.row] as? ALMessage {
+                    showDeleteAlert(conversation: conversation)
+                }
+            }
+        }
+    }
+    
+    func showDeleteAlert(conversation: ALMessage) {
+        let alert = UIAlertController(title: "Delete Conversation", message: "Do you want to delete the conversation?", preferredStyle: .alert)
+
+        let cancelButton = UIAlertAction(
+            title: localizedString(
+                forKey: "ButtonCancel",
+                withDefaultValue: SystemMessage.ButtonName.Cancel,
+                fileName: localizedStringFileName
+            ),
+            style: .cancel,
+            handler: nil
+        )
+        
+       let buttonTitle =  localizedString(
+            forKey: "DeleteButtonName",
+            withDefaultValue: "Delete",
+            fileName: localizedStringFileName
+        )
+        let deleteButton = UIAlertAction(title: buttonTitle, style: .destructive, handler: { [weak self] _ in
+            guard let weakSelf = self, ALDataNetworkConnection.checkDataNetworkAvailable() else { return }
+            let alert = weakSelf.displayAlert(viewController: weakSelf)
+            let messageService = ALMessageService()
+            if conversation.isGroupChat {
+                messageService.deleteMessageThread(nil, orChannelKey: conversation.groupId, withCompletion: {
+                    _, error in
+                    alert.dismiss(animated: false)
+                    guard error == nil else { return }
+                    let channelDbService = ALChannelDBService()
+                    channelDbService.deleteChannel(conversation.groupId)
+                    weakSelf.viewModel.remove(message: conversation)
+                    weakSelf.tableView.reloadData()
+                    DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) {
+                        let alert = UIAlertController(title: nil, message: "Successfully deleted the conversation!!", preferredStyle: .alert)
+                        let okButton = UIAlertAction(title: "Okay", style: .default, handler: nil)
+                        alert.addAction(okButton)
+                        weakSelf.present(alert, animated: true, completion: nil)
```

Comment:
Divide this function into two parts: one for showing the alert and other for handling the action.

Reply:
by default it will close the alert. created new function to show alerts. which can be used in multiple places

Reply:
I mean move the action block to a separate function.

Reply:
done